### PR TITLE
Fix #12617: zig fmt removing module comments

### LINKED ISSUE #12617: zig fmt: module comments are removed when a container lacks a trailing comma on it's last field
### Zig Version

0.10.0-dev.3685+dae7aeb33

### Steps to Reproduce

zig fmt -stdin < bug.zig

```zig
const Signature = enum(u32) { //! this should stay
_ };
```

### Expected Behavior

```zig
const Signature = enum(u32) { 
    //! this should stay
    _,
};
```

### Actual Behavior

```zig
const Signature = enum(u32) { _ };
```

### PR DESCRIPTION
**Pull request by yujiri8**
Is there a test suite for zig-fmt?

### CODE REVIEW

**Comment by Vexu** [comment]
> Is there a test suite for zig-fmt?

Yes, in `lib/std/zig/parser_test.zig`.

**Comment by yujiri8** [comment]
Damn. It failed.

**Comment by Vexu** [comment]
Note that line comments are not added to `tree.tokens`.

**Comment by yujiri8** [comment]
Ah okay, I think I see what really needs to be done. I can't test it right now because building zig takes so much ram that I have to kill other programs to do it, but I will come back to this later today and hopefully get this done for real.

**Comment by yujiri8** [comment]
I did it! All the tests pass, including a new one I added based on the issue.

File: lib/std/zig/render.zig [hunk 1]
```diff
@@ -1926,13 +1926,15 @@ fn renderContainerDecl(
 
     const src_has_trailing_comma = token_tags[rbrace - 1] == .comma;
     if (!src_has_trailing_comma) one_line: {
-        // We print all the members in-line unless one of the following conditions are true:
```
Comment (ifreund) [inline lib/std/zig/render.zig:1929]: There's no reason to remove this comment.

File: lib/std/zig/render.zig [hunk 2]
```diff
@@ -1926,13 +1926,15 @@ fn renderContainerDecl(
...
 
         // 1. The container has comments or multiline strings.
         if (hasComment(tree, lbrace, rbrace) or hasMultilineString(tree, lbrace, rbrace)) {
             break :one_line;
         }
 
+        // 2. The container has a container comment.
```
Comment (ifreund) [inline lib/std/zig/render.zig:1935]: Nit: using `// 2.` for the comment here means that the following comments are off-by one and should be updated.

**Review by ifreund — COMMENTED** [review]
Nice work!

**Comment by yujiri8** [comment]
Ugh, jeez... I had updated the numbers and undid and redid stuff and just :facepalm: :facepalm: :facepalm:

**Comment by yujiri8** [comment]
And I messed it up again when updating it just now :facepalm: :facepalm: :facepalm: Updated again

**Review by ifreund — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**